Sts1CobcSw/Hal/Gpio.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <Sts1CobcSw/Hal/PinNames.hpp>
+#include <type_safe/types.hpp>
+
+#include <rodos.h>
+
+#include <cstdint>
+
+namespace sts1cobcsw::hal
+{
+/**
+ * @brief The pin types of a GPIO pin.
+ */
+enum class PinType
+{
+    input,
+    output
+};
+
+/**
+ * @brief A wrapper for the init function of a GPIO pin. Only allows for single pins to be
+ * initialized.
+ *
+ * @param pin The pin to be initialized
+ * @param pinType The pin type (input/output)
+ * @param initVal The initial value for output pins
+ *
+ * @returns Returns 0 on success, -1 on failure
+ */
+auto InitPin(RODOS::HAL_GPIO & pin, PinType pinType, type_safe::bool_t initVal) -> int32_t;
+
+/**
+ * @brief A wrapper for the setPins function of a GPIO pin.
+ *
+ * @param pin The pin to be set
+ * @param pinVal The value to be set
+ */
+void SetPin(RODOS::HAL_GPIO & pin, type_safe::bool_t pinVal);
+
+/**
+ * @brief A wrapper for the readPins function of a GPIO pin.
+ *
+ * @param pin The pin to be read
+ *
+ * @returns true if the pin is set high, false otherwise
+ */
+auto ReadPin(RODOS::HAL_GPIO & pin) -> type_safe::bool_t;
```

Comment:
`the object type has qualifiers that are not compatible with the member function "RODOS::HAL_GPIO::readPins"` for `auto pinValue = pin.readPins();`.

Reply:
Yeah, the Rodos people don't know how to write const-correct code unfortunately. Reading the value of a pin should be possible with a `GPIO const &`, because semantically there is no reason for the GPIO class to change its state when calling that function. Maybe the cache the read value and therefore made the read function non-const but that is not correct – they should have made the cache value `mutable`. Const correctness is not about syntax but semantics.